### PR TITLE
Mongodb/ipv6

### PR DESCRIPTION
**Pull request by guptaarvindk**
**What this PR does / why we need it**:
suport for: 
- headless services and static clusterIP
- option to disable the IPV6 even in standalone deploymnet.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
certain kubernetes deployment need headless services and may not have IPV6 enabled.

### CODE REVIEW

**Comment by carrodher** [comment]
Before continue, can you take a look at the test errors?
https://circleci.com/gh/helm/charts/26080?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

File: stable/mongodb/templates/svc-primary-rs.yaml [hunk 1]
```diff
@@ -14,6 +14,11 @@ metadata:
 {{- end }}
 spec:
   type: {{ .Values.service.type }}
+   {{- if eq .Values.service.type "ClusterIP" }}
```
Comment (carrodher) [inline stable/mongodb/templates/svc-primary-rs.yaml]: Not tested, but... Can you try using it in a single line? Something like
```yaml
{{ if (.Values.service.clusterIP) and eq .Values.service.type "ClusterIP" }}
clusterIP: {{ .Values.service.clusterIP }}
{{- end }}

```

File: stable/mongodb/templates/svc-standalone.yaml [hunk 2]
```diff
@@ -14,6 +14,11 @@ metadata:
 {{- end }}
 spec:
   type: {{ .Values.service.type }}
+  {{- if eq .Values.service.type "ClusterIP" }}
```
Comment (carrodher) [inline stable/mongodb/templates/svc-standalone.yaml]: Same as above

File: stable/mongodb/templates/deployment-standalone.yaml [hunk 3]
```diff
@@ -78,6 +78,12 @@ spec:
         {{- end }}
         - name: MONGODB_EXTRA_FLAGS
           value: {{ default "" .Values.mongodbExtraFlags | join " " }}
+        - name: MONGODB_ENABLE_IPV6
```
Comment (carrodher) [inline stable/mongodb/templates/deployment-standalone.yaml:81]: What happens if you don't set this variable? By default, the behavior should be the same that use `"no"`, right? So, can you try something like the following code block?
```yaml
{{- if .Values.mongodbEnableIPv6 }}
- name: MONGODB_ENABLE_IPV6
  value: "yes"
{{- end }}
```

Comment (guptaarvindk) [reply stable/mongodb/templates/deployment-standalone.yaml:81]: I believe default value is yes. I followed the exact code as in statefulset-primary-rs.yaml, it was missing in standalone deployment. If we changing the way you suggested probably we should change in all the places.

File: stable/mongodb/templates/svc-primary-rs.yaml (hunk 1 above)
Comment (guptaarvindk) [reply stable/mongodb/templates/svc-primary-rs.yaml]: make sense will fix in next rev

**Comment by guptaarvindk** [comment]
Thanks for review @carrodher , can you please see my comments.

File: stable/mongodb/templates/deployment-standalone.yaml (hunk 3 above)
Comment (carrodher) [reply stable/mongodb/templates/deployment-standalone.yaml:81]: Perfect, we can keep it as it is

**Comment by carrodher** [comment]
Also, take a look at the version in the `Chart.yaml`, it seems a new version appears and you should resolve the conflicts

**Comment by carrodher** [comment]
You increased `appVersion` instead of `version`, that is the reason the CI is failing

File: stable/mongodb/templates/svc-primary-rs.yaml (hunk 1 above)
Comment (guptaarvindk) [reply stable/mongodb/templates/svc-primary-rs.yaml]: Hi @carrodher unfortunately what you suggested is not working in all scenario. I think probably thats why I went with two step process earlier as well. Do you have any better suggestion to make it work?

I your code above if first condition is satisfying and second condition is not, still its allowing to go inside the control. Not sure why, but doesn't seems to work.

I tired few more pipelines to combine two condition, but is failing in one or other way.

Is it ok to keep it in 2 step?

Comment (guptaarvindk) [reply stable/mongodb/templates/svc-primary-rs.yaml]: Hi @carrodher can you please confirm and approve the pull request.

Comment (carrodher) [reply stable/mongodb/templates/svc-primary-rs.yaml]: Hi, I am almost sure we can do it in a single if. Above solution was not testet and was to give an idea (pseudo-code), the real solution should be something like
```suggestion
   {{- if and (.Values.service.clusterIP) (eq .Values.service.type "ClusterIP") }}
```
or
```suggestion
   {{- if (and (.Values.service.clusterIP) (eq .Values.service.type "ClusterIP")) }}
```

Can you try it?

File: stable/mongodb/values-production.yaml [hunk 4]
```diff
@@ -73,6 +73,7 @@ clusterDomain: cluster.local
 service:
   annotations: {}
   type: ClusterIP
+  #clusterIP: None
```
Comment (carrodher) [inline stable/mongodb/values-production.yaml]: Sorry for being picky, can you add a space between the `#` and `clusterIP`? to be consistent with other values:
```suggestion
  # clusterIP: None
```

**Comment by carrodher** [comment]
It LGTM, please take a look at my last comment

/ok-to-test

File: stable/mongodb/values-production.yaml (hunk 4 above)
Comment (guptaarvindk) [reply stable/mongodb/values-production.yaml]: done. Thanks,

**Comment by guptaarvindk** [comment]
No worries @carrodher , thanks for review. Fixed that minor issue as well . Can you please take a look.

**Pull request merged**